queries/orderbook/batch_rewards.sql

```diff
@@ -0,0 +1,80 @@
+
+     reward_data AS (SELECT
+                       -- observations
+                       tx_hash,
+                       coalesce(
+                               solver,
+                         -- This is the winning solver (i.e. last entry of participants array)
+                               participants[array_length(participants, 1)]
+                         )                                    as solver,
+                       -- Right-hand terms in coalesces below represent the case when settlement
+                       -- observations are unavailable (i.e. no settlement corresponds to reported scores).
+                       -- In particular, this means that surplus, fee and execution cost are all zero.
+                       -- When there is an absence of settlement block number, we fall back
+                       -- on the block_deadline from the settlement_scores table.
+                       coalesce(block_number, block_deadline) as block_number,
+                       coalesce(execution_cost, 0)            as execution_cost,
+                       coalesce(surplus, 0)                   as surplus,
+                       coalesce(fee, 0)                       as fee,
+                       surplus + fee - reference_score        AS payment,
+                       -- scores
+                       winning_score,
+                       reference_score,
+                       -- participation
+                       participants
+                     FROM settlement_scores ss
+                            -- If there are reported scores,
+                            -- there will always be a record of auction participants
+                            JOIN auction_participants ap
+                                 ON ss.auction_id = ap.auction_id
+                       -- outer joins made in order to capture non-existent settlements.
+                            LEFT OUTER JOIN observed_settlements os
+                                            ON os.auction_id = ss.auction_id),
+
+     reward_per_tx as (SELECT tx_hash,
+                              solver,
+                              execution_cost,
+                              surplus,
+                              fee,
+                              surplus + fee - reference_score as reward_eth,
+                              winning_score,
+                              reference_score,
+                              participants                    as participating_solvers
+                       FROM reward_data),
+     participation_data as (select tx_hash, unnest(participating_solvers) as participant
+                            from reward_per_tx),
+     participation_counts as (select participant as solver, count(*) as num_participating_batches
+                              from participation_data
+                              group by participant),
+     primary_rewards as (select rpt.solver, sum(reward_eth) as total_reward_eth
```

Comment:
Somewhere between here and the definition of `reward_eth` (or `payment` or `uncapped_payment`) the cap needs to be applied.

We also need to keep track of the total transaction costs. On the one hand because of how the payment is split into COW and ETH, and on the other hand because of how the rewards budget is defined. We have
`total primary rewards = total primary payment - total transaction costs`
and
`total secondary rewards = total rewards budget - total primary rewards`.

When I am thinking about rewards in the new scheme, I am thinking in terms of `uncapped_payment = surplus + fee - reference_score`, `payment = cap( uncapped_payment )`, `gas_reimbursement = execution_cost * success` (this should be equal to what you call `execution_cost` with the convention of using 0 in case of a revert or failed deadline), `rewards = payment - gas_reimbursement`.

Reply:
Yes! Thanks for reminding me! We will need to introduce the capping before aggregating per solver. In fact I think we can introduce the capping inside the `sum(reward_eth)` since we still have access to the execution cost here (and we will parameterize the "EPSILON"). We can also sum up the total execution cost in this very same table.

Also, The number of participations (histogram) should be sufficient information for us to deduce the secondary distribution within our script (Something I forgot to mention in PR description and code commentary).

Great point!

Reply:
Alright @fhenneke - I have addressed the first two items here

- compute both capped reward
- keep also the total execution cost

These "capped rewards" have all bee computed per transaction. However, I am not sure what is up with the last stuff you mentioned and it almost seems like you are suggesting the capping be performed on the totals (rather than per transaction). Or....  maybe it is about the execution costs for failed settlements. I will quote the last paragraph here and continue with my "interpretation" below....

> When I am thinking about rewards in the new scheme, I am thinking in terms of uncapped_payment = surplus + fee - reference_score, payment = cap( uncapped_payment ), gas_reimbursement = execution_cost * success (this should be equal to what you call execution_cost with the convention of using 0 in case of a revert or failed deadline), rewards = payment - gas_reimbursement.


It looks a bit like the meaningful content here that has not already been addressed above is the total execution cost for successful settlements only. Based on the knowledge that there are no settlement events for failed transactions, we will write this query in a way (using something like "inner join") so that we do no pick up execution costs from failed transactions. I would hope that the query captures this already so that the "total_execution_cost" we see in the query already is actually the total successful execution cost. Since none of the tables exist yet in the DB its kinda tricky to test this, but we will circle back and include tests to ensure that this is indeed the case.